### DISCUSSION TITLE
RFC: cQube Architecture evolution and roadmap

### DISCUSSION THREAD

**Discussion by SreenivasNimmagadda** [Sunbird-cQube/community#79] [General]
**This post is created to request the community to provide me the feedback and suggestions of changes.** 

The cQube architecture road map was prepared in a document which available. Here’s a summary of proposed cQube latest architecture which is designed from the past experiences and cQube’s evolution from an architect’s perspective for the next version (cQube 3). 

![image (3)](https://user-images.githubusercontent.com/61133774/186157924-a76b5ee8-0f98-4e38-8468-cad889ebefd7.png)

For the full history of the architecture evaluation, Please refer [the architecture document ](https://docs.google.com/document/d/1fzCHoJHrBVc1Xycvq0dCRkyJ4twP0-uSKYqHnHJqjZk/edit?usp=sharing). This document defines the architectural changes that are happened so far in the cQube development. 

Explained the drawbacks of the older versions and the features of the newer version of the architecture in the points. 
We would like the community to review the architectural changes, share feedback and engage with us so that we can collaborate and make the architecture robust and useful for all adopters of cQube. 

@TLSravani @krishnakr6 @pandutibil @Sharathnaik38 Thanks for the inputs in preparation of the architecture preparation.

@pramodkvarma @surendrasinghs @alok-os @abhineetsingh10 @Ikpreet-Singh @SanthoshVasabhaktula @sharathkashyap @vrayulu @karsood

**Comment by surendrasinghs** [comment]
Thanks, @SreenivasNimmagadda. I have started adding comments to the document. Let's schedule a call soon to discuss feedback from everyone.

**Reply by pramodkvarma** [reply]
This is fantastic @SreenivasNimmagadda and team. I will surely look through the doc and provide any additional inputs. Thanks.
